mmdet/apis/det_inferencer.py

```diff
@@ -0,0 +1,353 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+import os.path as osp
+import warnings
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import mmcv
+import mmengine
+import numpy as np
+import torch.nn as nn
+from mmengine.dataset import Compose
+from mmengine.infer.infer import BaseInferencer, ModelType
+from mmengine.runner import load_checkpoint
+from mmengine.structures import InstanceData
+from mmengine.visualization import Visualizer
+
+from mmdet.registry import DATASETS, MODELS
+from mmdet.utils import ConfigType, register_all_modules
+from ..evaluation import get_classes
+
+InstanceList = List[InstanceData]
+InputType = Union[str, np.ndarray]
+InputsType = Union[InputType, Sequence[InputType]]
+PredType = Union[InstanceData, InstanceList]
+ImgType = Union[np.ndarray, Sequence[np.ndarray]]
+ResType = Union[Dict, List[Dict], InstanceData, List[InstanceData]]
+
+
+class MMDetInferencer(BaseInferencer):
```

Comment:
Without specific needs, the Inferencer should be named in task types. Here are some candidates to be discussed: [`DetInferencer`, `DetectionInferencer`, `ObjDetInferencer` or `ObjectDetectionInferencer`]